internal/cortex/frontend/transport/utils/utils.go

```diff
@@ -92,19 +88,20 @@ func (f *FailedQueryCache) updateFailedQueryCache(err error, queryExpressionNorm
 
 func (f *FailedQueryCache) addCacheEntry(queryExpressionNormalized string, queryExpressionRangeLength int) {
 	// Checks if queryExpression is already in cache, and updates time range length value to min of stored and new value.
-	if contains, _ := f.lruCache.ContainsOrAdd(queryExpressionNormalized, queryExpressionRangeLength); contains {
+	if contains := f.lruCache.Contains(queryExpressionNormalized); contains {
 		if oldValue, ok := f.lruCache.Get(queryExpressionNormalized); ok {
-			queryExpressionRangeLength = min(queryExpressionRangeLength, oldValue.(int))
+			queryExpressionRangeLength = min(queryExpressionRangeLength, oldValue)
 		}
-		f.lruCache.Add(queryExpressionNormalized, queryExpressionRangeLength)
 	}
+	f.lruCache.Add(queryExpressionNormalized, queryExpressionRangeLength)
```

Comment:
Can you confirm `Add()` is thread-safe? Multiple goroutines from the HTTP server may call this function here concurrently.

Reply:
yes it uses a mutex